components/AddSpot/index.js

```diff
@@ -0,0 +1,230 @@
+import styled from "styled-components";
+import React, { useState } from "react";
+import { useRouter } from "next/router";
+import mockData, { addSpotData } from "../../lib/mock-data";
+
+export default function NewSpotForm() {
+  const [spotName, setSpotName] = useState("NAME THIS SPOT HERE");
+  const [longitude, setLongitude] = useState("");
+  const [latitude, setLatitude] = useState("");
+  const [isFocused, setIsFocused] = useState(false);
+  const [isNameError, setIsNameError] = useState(false);
+  const [isDuplicateNameError, setIsDuplicateNameError] = useState(false);
```

Comment:
Wouldn´t it be possible to comprimise this states? 
i guess you could use an array of objects instead as initialvalue